common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleClient.java

```diff
@@ -60,10 +61,19 @@
   public ExternalShuffleClient(
       TransportConf conf,
       SecretKeyHolder secretKeyHolder,
-      boolean authEnabled) {
+      boolean authEnabled,
+      long registrationTimeoutMilli) {
     this.conf = conf;
     this.secretKeyHolder = secretKeyHolder;
     this.authEnabled = authEnabled;
+    this.registrationTimeoutMilli = registrationTimeoutMilli;
+  }
+
+  public ExternalShuffleClient(
```

Comment:
Do we actually need this backwards-compatible constructor? AFAIK this interface is internal only and shouldn't be used from outside of Spark.